### PR TITLE
Build files tree

### PR DESCRIPTION
**Pull request by sergesemashko**
Left some TODO's that would be nice to add to the code.

@scottnath please review

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -6,4 +6,10 @@ var dir = require('node-dir'),
 var createPatternObject = require('./lib/create-pattern-object.js'),
     PLUGIN_NAME = 'pattern-importer';
 
-console.log(createPatternObject('./app/_patterns'));
+var config = createPatternObject('../../app/_patterns', function (err, filesTree){
```
Comment (sergesemashko) [inline index.js]: I followed the instruction and pattern-presenter was installed in ```node_modules/```, so I corrected the path to app folder. Seems like this module should be ```app``` folder and path to patterns should be a ENV or just config variable. @scottnath what do you think?

File: lib/create-pattern-object.js [hunk 2]
```diff
@@ -1,69 +1,79 @@
...
+  gutil = require('gulp-util'),
+  PLUGIN_NAME = 'pattern-importer';
+
 /*
  * Creates an object from a folder of compiled patterns
  * @module createPatternObject
  *
  * @param {Object} patternFolder
-*/
-var createPatternObject = function createPatternObject (patternFolder) {
```
Comment (scottnath) [inline lib/create-pattern-object.js]: Why was this removed? If we don't have this variable, what do we reference for module.exports?

Comment (sergesemashko) [inline lib/create-pattern-object.js]: it was not necessary, if function has a name it will be accessible like a variable

File: index.js (hunk 1 above)
Comment (scottnath) [inline index.js]: I definitely think that the path to patterns should be a config variable - agreed.

for location of `pattern-presenter`, I have been working under the assumption in Atlas that everything in the `app/` folder would be project-specific items and generally items eventually headed to the web. So, any npm modules I was keeping in the main `node-modules` folder outside of `app` because they would be generalized functionality for Atlas and also don't work online. But items brought in via bower (like the pattern library repo, or maybe angularjs or whatever code the site would need in a browser) would be brought into the `app` folder.

this is also why I put in a separate gulp folder for the app (`app/_gulp`) where project-specific gulp tasks could be added as well as overriding the configuration file.

I figured this separation would allow the app we're building to be separated from the tool we're building it with. And thus, if anything went wrong with the tool and it didn't seem to be working - a developer could do a fresh Atlas install and drop the app folder into the fresh install and be back at work.

rebuttal?

File: lib/create-pattern-object.js [hunk 3]
```diff
@@ -1,69 +1,79 @@
...
+      filesTree.categories['uncategorized'].patterns.push( {name: fileData.name, url: fileData.source} );
+    }
+    next();
+  }
+
+
+  /* read through the folder and find all our compiled.yml files */
 
+  /**
+   * @TODO: put a promise to return here rather than execute callback
```
Comment (scottnath) [inline lib/create-pattern-object.js]: Agreed. Promises was where I got stuck last Thursday. Def needs to be a promise.

Is it possible to come up with a common way to do promises? I think the whole pattern system would be better-built if it had promises used throughout.

**Comment by scottnath** [comment]
Merging. No changes needed at this point.
